File: android/app/src/main/java/com/zahraag/pawsitivehabits/data/models/Pet.kt
package com.zahraag.pawsitivehabits.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
@Entity(tableName ="pet_table")
data class Pet(
    @PrimaryKey @SerializedName("_id")
    val id: String = org.bson.types.ObjectId().toHexString(),
    val serverId: String? = null,
    var userId: String,
    var name: String,
    var gender: String,
    var petType: String,
    var breed: String? = null,
    var dateOfBirth: Long,
    var adoptionDate: Long,
    var microchipId: String? = null,
    var isNeutered: Boolean = false,
    var localImagePath: String? = null,
    var remoteImageUrl: String? = null,
    var colour: String?=null,
    var notes: String? = null,
    var customColour: String? = "#FF5733",

    @kotlinx.serialization.Transient
    var isSynced: Boolean = false
)

File: api/src/models/pet.js
const mongoose = require('mongoose');

const petSchema = new mongoose.Schema({
  _id: { type: String },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  name: { type: String, required: true,trim: true },
  gender: { type: String, required: true },
  petType: { type: String, required: true },
  breed: { type: String, default: null,trim: true },
  dateOfBirth: { type: Number, required: true },
  adoptionDate: { type: Number, required: true },
  microchipId: { type: String, default: null,trim: true },
  isNeutered: { type: Boolean, default: false },
  imageUrl: { type: String, default: null },
  localImageUrl: { type: String, default: null },
  remoteImageUrl: { type: String, default: null },
  colour: { type: String, default:null,trim: true},
  notes: { type: String, default:null},
  customColour: { type: String, default: null }
}, { timestamps: true });

module.exports = mongoose.model('Pet', petSchema);